megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -2157,6 +2170,172 @@ private void launchFighters(MovePath path) {
         }
     }
 
+    /**
+     * Sets the mode for AMS on each unit this bot controls. This may be on or off, and possibly
+     * manual fire if the game options allow it, with any change taking effect next round.
+     * Normal setting is to have the AMS active/automatic. It may be turned off to conserve
+     * ammo on relatively undamaged units, and laser AMS may be turned off to help reduce
+     * overheating. Manual use is reserved as an emergency anti-infantry measure.
+     *
+     */
+    private void setAMSModes() {
+
+        // Get conventional infantry if manual mode is available
+        List<Entity> enemyInfantry = new ArrayList<>();
+        if (game.getOptions().booleanOption(OptionsConstants.ADVCOMBAT_TACOPS_MANUAL_AMS)) {
+            for (Entity curEnemy : this.getEnemyEntities().stream().filter(Entity::isDeployed).collect(Collectors.toSet())) {
+                if (curEnemy.getPosition() != null && curEnemy.isVisibleToEnemy()) {
+
+                    if (curEnemy instanceof Infantry && !(curEnemy instanceof EjectedCrew)) {
+                        enemyInfantry.add(curEnemy);
+                    }
+
+                }
+            }
+        }
+
+        for (Entity curEntity : this.getEntitiesOwned()) {
+            if (!curEntity.isDeployed() || curEntity.getPosition() == null) {
+                continue;
+            }
+
+            List<WeaponMounted> activeAMS = curEntity.
+                    getWeaponList().
+                    stream().
+                    filter(w -> w.getType().hasFlag(AmmoWeapon.F_AMS) && w.hasModes()).
+                    collect(Collectors.toList());
+
+            if (!activeAMS.isEmpty()) {
+
+                // Set default to on/automatic and test to see if it should be off or manual instead
+                EquipmentMode newAMSMode = EquipmentMode.getMode(Weapon.MODE_AMS_ON);
+
+                boolean isOverheating = (curEntity instanceof Mech) && (curEntity.getHeat() >= 14);
```

Comment:
This heat threshold number should either be:
A) a constant declared at the top of the file, or
B) a value passed in to the function / read from some member, or
C) a reference to any other existing heat limit variable already used by Princess
so we don't have conflicting behavior.

Reply:
Yeah, ideally it would tie into the existing code that handles heat.  The big problem is the existing code is during the movement/weapons fire phases, and like the stealth armor control this is happening during the end phase (which is why it can have a look at the heat level rather than trying to predict it).  So all that fire control objects/information is no longer valid or available.

I'll put in a constant for now.  I would like something a little more... dynamic, I guess, for what constitutes 'overheating'.  But keeping the complexity down seems to be the better option to start with.

Reply:
> Yeah, ideally it would tie into the existing code that handles heat. The big problem is the existing code is during the movement/weapons fire phases, and like the stealth armor control this is happening during the end phase (which is why it can have a look at the heat level rather than trying to predict it). So all that fire control objects/information is no longer valid or available.
> 
> I'll put in a constant for now. I would like something a little more... dynamic, I guess, for what constitutes 'overheating'. But keeping the complexity down seems to be the better option to start with.

I might be misremembering, but I think there's a heat limit value that's used in the "with/without overheating" firing plan code that might be usable.  IIRC we calculate something based on aggression level; if it's not immediately available, that might be nice to pull out to a method or instance member.